tests/openwrt/test_interfaces_dsa.py

```diff
@@ -1145,6 +1145,172 @@ def test_parse_l2_options_interface(self):
 
+    _vlan_filtering_bridge_netjson = {
+        "interfaces": [
+            {
+                "type": "bridge",
+                "bridge_members": ["lan1", "lan2", "lan3"],
+                "name": "br-lan",
+                "network": "home_vlan",
+                "vlan_filtering": [
+                    {
+                        "vlan": 1,
+                        "ports": [
+                            {"ifname": "lan1", "tagging": "t", "primary_vid": True},
+                            {"ifname": "lan2", "tagging": "t"},
+                        ],
+                    },
+                    {
+                        "vlan": 2,
+                        "ports": [
+                            {"ifname": "lan1", "tagging": "t", "primary_vid": False},
+                            {"ifname": "lan3", "tagging": "u", "primary_vid": True},
+                        ],
+                    },
+                ],
+            }
+        ]
+    }
+
+    _vlan_filtering_bridge_uci = """package network
+
+config device 'device_home_vlan'
+    option name 'br-lan'
+    list ports 'lan1'
+    list ports 'lan2'
+    list ports 'lan3'
+    option type 'bridge'
+    option vlan_filtering '1'
+
+config bridge-vlan 'home_vlan_1'
+    option device 'br-lan'
+    list ports 'lan1:t*'
+    list ports 'lan2:t'
+    option vlan '1'
+
+config bridge-vlan 'home_vlan_2'
+    option device 'br-lan'
+    list ports 'lan1:t'
+    list ports 'lan3:u*'
+    option vlan '2'
+
+config interface 'home_vlan_1'
+    option device 'br-lan.1'
+    option proto 'none'
+
+config interface 'home_vlan_2'
+    option device 'br-lan.2'
+    option proto 'none'
+
+config interface 'home_vlan'
+    option device 'br-lan'
+    option proto 'none'
+"""
+
+    def test_render_bridge_vlan_filtering(self):
+        o = OpenWrt(self._vlan_filtering_bridge_netjson)
+        self.assertEqual(self._tabs(self._vlan_filtering_bridge_uci), o.render())
+
+        with self.subTest('Test setting PVID on same port on different VLANS'):
+            netjson = deepcopy(self._vlan_filtering_bridge_netjson)
+            netjson['interfaces'][0]['vlan_filtering'][1]['ports'][0][
+                'primary_vid'
+            ] = True
+            with self.assertRaises(ValidationError) as error:
+                OpenWrt(netjson).validate()
+            self.assertEqual(
+                error.exception.message,
+                (
+                    'Invalid configuration triggered by "#/interfaces/0"'
+                    ' says: Primary VID can be set only one VLAN for a port.'
+                ),
+            )
+
+    def test_parse_bridge_vlan_filtering(self):
+        o = OpenWrt(native=self._vlan_filtering_bridge_uci)
+        expected = deepcopy(self._vlan_filtering_bridge_netjson)
+        expected['interfaces'][0]['vlan_filtering'][0]['ports'][1][
+            'primary_vid'
+        ] = False
+        expected['interfaces'][0]['name'] = 'br-home_vlan'
```

Comment:
After allowing user to set the logical name for VLAN interface, the parsing of the bridge VLAN breaks.